shared/language-specs/comments.ts

```diff
@@ -1,35 +1,46 @@
-import { CommentStyle } from './spec'
+import { BlockCommentStyle, CommentStyle } from './spec'
 
-export const cStyleLineNoiseRegex = /(^\s*\*\s?)?/
+/** Matches two or more slashes followed by one optional space. */
+export const slashPattern = /\/\/\/*\s?/
 
-export const cStyleBlock: any = {
+/** Matches three slashes followed by one optional space. */
+export const tripleSlashPattern = /\/\/\/\s?/
+
+/** Matches a hash followed by one optional space. */
+export const hashPattern = /#\s?/
+
+/** Matches two or more dashes followed by one optional space. */
+export const dashPattern = /---*\s?/
+
+// TODO - does not need to be optional?
```

Comment:
I wondered about this too. I would have expected this to be something like `^\s*(\*\s?)?`, so that it will match and consume star noise if it's present but also accept lines (given to be inside a block comment) that lack star noise.

I wasn't able to convince myself I wouldn't break something by changing it, but I think we only ever use it for matching rather than trimming, so it is _probably_ fine.

Reply:
Will look into this a bit more.